runtime/error.c

```diff
@@ -148,13 +165,66 @@ void _bal_error_backtrace_print(ErrorPtr ep) {
     fflush(stderr);
 }
 
+static void processPCs(GC PC* pcs, uint32_t nPCs, uint32_t start, int64_t lineNumber, BacktraceError *error) {
+    // Handle the starting pc seperately
+    processInitialPC(pcs[start], lineNumber, error);
+    if (error->code == NO_DEBUG_INFO) {
+        for (uint32_t i = start + 1; i < nPCs; i++) {
+            // We do not need to pass error_callback because, already we have an error.
+            // Also we do not need to pass data because we do not use it
+            backtrace_syminfo(state, pcs[i], printMissingSymbols, NULL, NULL);
+        }
+    }
+    else {
+        for (uint32_t i = start + 1; i < nPCs; i++) {
+            backtrace_pcinfo(state, pcs[i], printBacktraceLine, storeBacktraceError, error);
+        }
+    }
+}
+
+static void processInitialPC(PC pc, int64_t lineNumber, BacktraceError *error) {
+    BacktraceStartLine backtraceStartLine = { { error->code, error->message }, lineNumber };
+    backtrace_pcinfo(state, pc, printBacktraceStartLine, storeBacktraceError, &backtraceStartLine);
+    if (backtraceStartLine.error.code == NO_DEBUG_INFO) {
+        // We do not need to pass error_callback because, already we have an error
+        backtrace_syminfo(state, pc, printInitialMissingSymbols, NULL, &backtraceStartLine);
+    }
+    error->code = backtraceStartLine.error.code;
+    error->message = backtraceStartLine.error.message;
+}
+
 // Implementation of backtrace_full_callback
 static int printBacktraceLine(void *data, PC pc, const char *filename, int lineno, const char *function) {
+    printLine(filename, lineno, function);
+    return SUCCESS;
+}
+
+// Implementation of backtrace_full_callback
+static int printBacktraceStartLine(void *data, PC pc, const char *filename, int lineno, const char *function) {
+    BacktraceStartLine *backtraceStart = data;
+    printLine(filename, backtraceStart->lineNumber, function);
+    return SUCCESS;
+}
+
+// Implementation of backtrace_syminfo_callback
+static void printMissingSymbols(void *data, uintptr_t pc, const char *symname, uintptr_t symval, uintptr_t symsize) {
+    printLine(NULL, 0, symname);
+}
+
+// Implementation of backtrace_syminfo_callback
+static void printInitialMissingSymbols(void *data, uintptr_t pc, const char *symname, uintptr_t symval, uintptr_t symsize) {
+    BacktraceStartLine *backtraceStartLine = data;
+    printLine(NULL, backtraceStartLine->lineNumber, symname);
+}
+
+static void printLine(const char *filename, int64_t lineNumber, const char *function) {
     if (function != NULL && filename != NULL) {
```

Comment:
You should check separately for function/filename/lineno being NULL/0, like I did here:

https://github.com/ballerina-platform/nballerina/blob/5f4da8c8ebd0d26f39984ee00faca7c7d13561a5/runtime/error.c#L151

Reply:
I used this implementation and did a small change. I changed `%d` to `%ld`, because the type of lineNumber passed from _bal_panic_construct is int64_t

Reply:
`%ld` is right if you are using `long`. int64_t uses PRi64

Reply:
Fixed